File: TESTDB/src/data/Play.java
package data;


import javax.persistence.*;
import java.util.HashMap;
import java.util.Map;

@Entity
public class Play {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private boolean bored = false;
    @Transient
    private boolean football = false;
    @Transient
    private Map<String, Boolean> map = new HashMap<>();

    Play() {
    }

    boolean isBored() {
        return bored;
    }

    private void setBored(boolean bored) {
        this.bored = bored;
    }

    public int getId() {
        return id;
    }

    private void setId(int id) {
        this.id = id;
    }

    boolean isFootball() {
        return football;
    }

    private void setFootball(boolean football) {
        this.football = football;
    }

    Map<String, Boolean> getNeeds() {
        if (!isBored()) {
            map.put("Börje is bored", isBored());
        }
        if (!isFootball()) {
            map.put("Börje wants to play football", isFootball());
        }
        return map;
    }
}

File: TESTDB/src/Dao/Play.kt
package Dao

import java.util.*
import javax.persistence.*


@Entity
class Play : Care.CareInformation, DaoMaster() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int = 0

    /**
     * Gets set using reflection api
     * @see DaoMaster
     */
    private var hockey = false
    private var football = false

    @Transient
    private var unicornProperties = LinkedHashMap<String, Boolean>()


    override fun newUnicorn(): Play {
        unicornProperties = setNewUnicorn(unicornProperties)
        return this
    }

    /**
     * Adds false values to the unicornProperties
     */
    override fun checkForUpdates() {
        unicornProperties = checkAndReturnUpdates(unicornProperties)
    }

    override fun save(key: String): String {
        unicornProperties.remove(saveChanges(key))
        return "enjoyed playing $key"
    }

    override fun getCategories(): LinkedHashMap<String, Boolean> {
        return unicornProperties
    }

    override fun toString(): String {
        return javaClass.simpleName
    }
}